### PR TITLE
use ibc_proto::protobuf::Protobuf to replace tendermint_proto::Protobuf.

### LINKED ISSUE #747: Tendermint `ConsensusState -> Any` can crash if out of memory
[Here](https://github.com/cosmos/ibc-rs/blob/cd3a266cf4dda173a0c1797d84d984e3b36c3c19/crates/ibc/src/clients/ics07_tendermint/consensus_state.rs#L128). 

This probably involves updating tendermint-rs's `Protobuf` in a similar way that was done for the `Protobuf` in ibc-proto-rs (which removed the `Result` return for `encode_vec()`).

### PR DESCRIPTION
**Pull request by DaviRain-Su**
Closes: #747 

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/core/ics03_connection/handler/conn_open_ack.rs [hunk 1]
```diff
@@ -117,9 +117,7 @@ where
                 &msg.proof_consensus_state_of_a_on_b,
                 consensus_state_of_b_on_a.root(),
                 Path::ClientConsensusState(client_cons_state_path_on_b),
-                expected_consensus_state_of_a_on_b
-                    .encode_vec()
-                    .map_err(ConnectionError::ConsensusStateEncodeFailure)?,
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics03_connection/handler/conn_open_ack.rs:122]: Would you remove the `ConsensusStateEncodeFailure` from the errors? It's no longer needed.

File: crates/ibc/src/core/ics02_client/consensus_state.rs [hunk 2]
```diff
@@ -26,7 +26,5 @@ pub trait ConsensusState: Send + Sync {
     /// first converting to the raw type (i.e. the protobuf definition), and then
     /// serializing that.
     ///
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics02_client/consensus_state.rs]: Can we remove this empty docstring line?

**Review by Farhad-Shabani — COMMENTED** [review]
Just a few nits, otherwise LGTM :pray:

File: .changelog/unreleased/bug/747-fix-747-Protobuf-out-of-memory.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- Tendermint ConsensusState -> Any can crash if out of memory
```
Comment (Farhad-Shabani) [inline .changelog/unreleased/bug/747-fix-747-Protobuf-out-of-memory.md:1]: Would you also add a unclog for the own PR under `breaking-changes`, so we can track the `encode_vec` signature change too?

File: crates/ibc/src/core/ics02_client/consensus_state.rs (hunk 2 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics02_client/consensus_state.rs]: fixed

File: crates/ibc/src/core/ics03_connection/handler/conn_open_ack.rs (hunk 1 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics03_connection/handler/conn_open_ack.rs:122]: fixed

File: .changelog/unreleased/bug/747-fix-747-Protobuf-out-of-memory.md (hunk 3 above)
Comment (DaviRain-Su) [reply .changelog/unreleased/bug/747-fix-747-Protobuf-out-of-memory.md:1]: fixed

**Review by Farhad-Shabani — APPROVED** [review]

**Pull request merged**